frontend/src/components/github-button.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import styled from 'styled-components';
+import PopupWindow from '../helpers/popup-window';
+import { toQuery } from '../helpers/url-utils';
+import { BaseButton } from './styled/base-button';
+
+const GitHubButton = ({ buttonText, onSuccess, onFailure }) => {
+  const githubOauth2Url = 'https://github.com/login/oauth/authorize';
+  const githubOauth2Params = {
+    client_id: process.env.REACT_APP_GITHUB_CLIENT_ID,
+    scope: 'user:email',
+  };
+
+  const handleClick = event => {
+    event.preventDefault();
+    const search = toQuery(githubOauth2Params);
+    new PopupWindow('github-auth', `${githubOauth2Url}?${search}`)
+      .then(response => onSuccess(response))
+      .catch(error => onFailure(error));
+  };
+
+  return (
+    <ButtonBody onClick={handleClick}>
+      {buttonText}
+      <svg aria-hidden="true" fill="currentColor" viewBox="0 0 20 20">
```

Comment:
better use inline svg components, but manual inline is very troublesome to support